### PR TITLE
[thread-cert] refactor case 6.1.3, 6.1.5, 6.1.7 and 6.1.9 using pktverify

### PR DESCRIPTION
**Pull request by FiveDimensions**

### CODE REVIEW

File: tests/scripts/thread-cert/Cert_5_1_07_MaxChildCount.py [hunk 1]
```diff
@@ -150,6 +167,48 @@ def test(self):
...
+        pkts = pv.pkts
+        pv.summary.show()
+
+        ROUTER = pv.vars['ROUTER']
+        router_pkts = pkts.filter_wpan_src64(ROUTER)
+
+        # Step 1: The DUT MUST send properly formatted MLE Parent Response
+        #         and MLE Child ID Response to each child.
+        for i in range(1, 7):
+            _pkts = router_pkts.copy().filter_wpan_dst64(pv.vars['SED%d' % i])
```
Comment (simonlingoogle) [inline tests/scripts/thread-cert/Cert_5_1_07_MaxChildCount.py:180]:
```suggestion
            _pkts = router_pkts.filter_wpan_dst64(pv.vars['SED%d' % i], cascade=False)
```
We can use `cascade=False` to elide `.copy()`

File: tests/scripts/thread-cert/Cert_5_1_07_MaxChildCount.py [hunk 2]
```diff
@@ -150,6 +167,48 @@ def test(self):
...
+            _pkts.filter_mle_cmd(MLE_PARENT_RESPONSE).must_next()
+            _pkts.filter_mle_cmd(MLE_CHILD_ID_RESPONSE).must_next()
+
+        for i in range(1, 5):
+            _pkts = router_pkts.copy().filter_wpan_dst64(pv.vars['MED%d' % i])
```
Comment (simonlingoogle) [inline tests/scripts/thread-cert/Cert_5_1_07_MaxChildCount.py:185]: ditto: use `cascade=False`

File: tests/scripts/thread-cert/Cert_5_1_07_MaxChildCount.py [hunk 3]
```diff
@@ -150,6 +167,48 @@ def test(self):
...
+            _pkts.filter('wpan.dst16 == {rloc16}',
+                         rloc16=leader_rloc16).filter_ping_reply(
+                             identifier=p.icmpv6.echo.identifier).must_next()
+
+        # Step 3: The DUT MUST properly forward ICMPv6 Echo Requests to all SED children
+        #         The DUT MUST properly forward ICMPv6 Echo Replies to the Leader
+        for i in range(1, 7):
+            rloc16 = pv.vars['SED%d_RLOC16' % i]
+            _pkts = router_pkts.copy()
+            p = _pkts.filter('wpan.dst16 == {rloc16}',
```
Comment (simonlingoogle) [inline tests/scripts/thread-cert/Cert_5_1_07_MaxChildCount.py:206]: `p` not used?

File: tests/scripts/thread-cert/Cert_6_1_05_REEDAttachConnectivity.py [hunk 4]
```diff
@@ -80,21 +87,62 @@ def test(self):
...
+
+        # Step 5: The DUT MUST send a MLE Child ID Request
+        _ed_pkts.filter_wpan_dst64(REED_1).filter_mle_cmd(
+            consts.MLE_CHILD_ID_REQUEST).must_next().must_verify(
+                lambda p: {4, 5, 8, 1, 2, 18, 19, 13} <= set(p.mle.tlv.type))
+
+        # Step 8: The DUT MUST respond with ICMPv6 Echo Reply
+        ed_mleid = pv.vars['ED_MLEID']
+        reed1_mleid = pv.vars['REED_1_MLEID']
+        _reed1_pkts.filter(lambda p: p.ipv6.src == reed1_mleid and p.ipv6.dst ==
```
Comment (simonlingoogle) [inline tests/scripts/thread-cert/Cert_6_1_05_REEDAttachConnectivity.py:141]: nit: use `filter_ipv6_src_dst`

File: tests/scripts/thread-cert/Cert_6_1_07_RouterAttachLinkQuality.py [hunk 5]
```diff
@@ -81,6 +87,39 @@ def test(self):
...
+
+    def verify(self, pv):
+        pkts = pv.pkts
+        pv.summary.show()
+
+        ROUTER_1 = pv.vars['ROUTER_1']
+        ED = pv.vars['ED']
+        _router1_pkts = pkts.filter_wpan_src64(ROUTER_1)
+        _ed_pkts = pkts.filter_wpan_src64(ED)
+
```
Comment (simonlingoogle) [inline tests/scripts/thread-cert/Cert_6_1_07_RouterAttachLinkQuality.py:103]: nit: use one filter if all steps are checked in a timely manner.

File: tests/scripts/thread-cert/Cert_6_1_09_EDSynchronization.py [hunk 6]
```diff
@@ -92,6 +97,35 @@ def test(self):
...
+        pv.summary.show()
+
+        ROUTER_1 = pv.vars['ROUTER_1']
+        ROUTER_2 = pv.vars['ROUTER_2']
+        ROUTER_3 = pv.vars['ROUTER_3']
+        ED = pv.vars['ED']
+        _router1_pkts = pkts.filter_wpan_src64(ROUTER_1)
+        _router2_pkts = pkts.filter_wpan_src64(ROUTER_2)
+        _router3_pkts = pkts.filter_wpan_src64(ROUTER_3)
+        _ed_pkts = pkts.filter_wpan_src64(ED)
```
Comment (simonlingoogle) [inline tests/scripts/thread-cert/Cert_6_1_09_EDSynchronization.py:111]: nit: use less filters.

File: tests/scripts/thread-cert/Cert_6_1_03_RouterAttachConnectivity.py [hunk 7]
```diff
@@ -91,10 +98,41 @@ def test(self):
...
+
+        # Step 2: The DUT MUST send a MLE Parent Request to the
+        # All-Routers multicast address
+        _ed_pkts.filter_mle_cmd(
+            consts.MLE_PARENT_REQUEST).must_next().must_verify(
+                lambda p: {1, 3, 14, 18} == set(p.mle.tlv.type
+                                               ) and p.mle.tlv.scan_mask.r == 1)
+
+        # Step 3: Router_2, Router_3 Respond with MLE Parent Response
+        # Step4: DUT Send a Child ID Request to Router_3 due to better connectivity
```
Comment (jwhui) [inline tests/scripts/thread-cert/Cert_6_1_03_RouterAttachConnectivity.py:123]:
```suggestion
        # Step 4: DUT Send a Child ID Request to Router_3 due to better connectivity
```

**Comment by jwhui** [comment]
@FiveDimensions , please rebase and resolve comments above.

**Comment by jwhui** [comment]
> @FiveDimensions , please rebase and resolve comments above.

@FiveDimensions , friendly reminder.

File: tests/scripts/thread-cert/Cert_6_1_05_REEDAttachConnectivity.py (hunk 4 above)
Comment (FiveDimensions) [reply tests/scripts/thread-cert/Cert_6_1_05_REEDAttachConnectivity.py]: done

File: tests/scripts/thread-cert/Cert_6_1_03_RouterAttachConnectivity.py (hunk 7 above)
Comment (FiveDimensions) [reply tests/scripts/thread-cert/Cert_6_1_03_RouterAttachConnectivity.py]: done

**Review by simonlingoogle — APPROVED** [review]
LGTM :+1:

File: tests/scripts/thread-cert/Cert_6_1_07_RouterAttachLinkQuality.py (hunk 5 above)
Comment (jwhui) [reply tests/scripts/thread-cert/Cert_6_1_07_RouterAttachLinkQuality.py:103]: @FiveDimensions , is this comment resolved?

File: tests/scripts/thread-cert/Cert_6_1_03_RouterAttachConnectivity.py [hunk 8]
```diff
@@ -91,10 +98,42 @@ def test(self):
...
+        pkts = pv.pkts
+        pv.summary.show()
+
+        ROUTER_3 = pv.vars['ROUTER_3']
+        ED = pv.vars['ED']
+        _router3_pkts = pkts.filter_wpan_src64(ROUTER_3)
+        _ed_pkts = pkts.filter_wpan_src64(ED)
+
+        # Step 2: The DUT MUST send a MLE Parent Request to the
+        # All-Routers multicast address
```
Comment (jwhui) [inline tests/scripts/thread-cert/Cert_6_1_03_RouterAttachConnectivity.py:116]: Are we validating the all-routers multicast address?

File: tests/scripts/thread-cert/Cert_6_1_05_REEDAttachConnectivity.py [hunk 9]
```diff
@@ -80,21 +87,60 @@ def test(self):
...
+        pkts = pv.pkts
+        pv.summary.show()
+
+        REED_1 = pv.vars['REED_1']
+        ED = pv.vars['ED']
+        _reed1_pkts = pkts.filter_wpan_src64(REED_1)
+        _ed_pkts = pkts.filter_wpan_src64(ED)
+
+        # Step 2: The DUT MUST send a MLE Parent Request to the
+        # All-Routers multicast address
```
Comment (jwhui) [inline tests/scripts/thread-cert/Cert_6_1_05_REEDAttachConnectivity.py:118]: Are we validating the all-routers multicast address?

File: tests/scripts/thread-cert/Cert_6_1_07_RouterAttachLinkQuality.py [hunk 10]
```diff
@@ -81,6 +87,39 @@ def test(self):
...
+        # Step 3: The DUT MUST send a MLE Parent Request to the
+        # All-Routers multicast address
```
Comment (jwhui) [inline tests/scripts/thread-cert/Cert_6_1_07_RouterAttachLinkQuality.py:105]: Are we validating the all-routers multicast address?

File: tests/scripts/thread-cert/Cert_6_1_07_RouterAttachLinkQuality.py (hunk 5 above)
Comment (FiveDimensions) [reply tests/scripts/thread-cert/Cert_6_1_07_RouterAttachLinkQuality.py]: resolved in commit 5

File: tests/scripts/thread-cert/Cert_6_1_03_RouterAttachConnectivity.py (hunk 8 above)
Comment (FiveDimensions) [reply tests/scripts/thread-cert/Cert_6_1_03_RouterAttachConnectivity.py:116]: yes, we should. Added in the commit 6

File: tests/scripts/thread-cert/Cert_6_1_05_REEDAttachConnectivity.py (hunk 9 above)
Comment (FiveDimensions) [reply tests/scripts/thread-cert/Cert_6_1_05_REEDAttachConnectivity.py:118]: yes, we should. Added in the commit 6

File: tests/scripts/thread-cert/Cert_6_1_07_RouterAttachLinkQuality.py (hunk 10 above)
Comment (FiveDimensions) [reply tests/scripts/thread-cert/Cert_6_1_07_RouterAttachLinkQuality.py:104]: yes, we should. Added in the commit 6

**Pull request merged**